src/streaming/controllers/BufferController.js

```diff
@@ -184,13 +186,17 @@ function BufferController(config) {
     }
```

Comment:
I've just noticed that the event stream stuff above no longer works in low latency mode and an exception is throw in `handleInbandEvents` if a stream signals it contains inband events. This is because `getRequests` for state `FRAGMENT_MODEL_EXECUTED` will return nothing until the request has actually completed (ie the whole segment has arrived), and therefore `request` is undefined. The problem seems to arise from onMediaFragmentLoaded is called for incomplete chunks of data.

Perhaps there will need to be some box handling in the downloader/elsewhere after all, since the UA is not capable of handling this box - it must be done in JS.

Reply:
Good point, @bbcrddave. @mmarciel, did in that way (box parsing using codem-isoboxer) in her first implementation because she detected some stability issues when working with not completed boxes. This was later changed to reduce the impact on performance.

I am now reviewing this part. In the worst case, in case there is no way of avoiding parsing, I think makes sense using a new and simplified "parser" focused on speed/perf for this, instead of using codem-isoboxer. We really don't need to parse boxes, just reading their lengths (and probably their types so we just append when finding specific box types) would be enough. It will be just parsing a few numbers and move though the data array which should be quite fast.

I will share my results.

Reply:
Apart from issues with events, I have been testing a lot (quite a lot) this in Chrome/Firefox and not inserting complete boxes also causes stability issues from time to time. Although most of the time works, there are situations in which playback is interrupted because a decoding issue.

As a solution I have added a new method to BoxParser.js which finds specific and completed top ISO boxes in a data array. It just iterates over top boxes and didn't parse anything so it is quite fast and I don't expect perceivable perf degradation because it. It is around 50-100x faster than parsing same chunks with ISOBox

Not a fair comparison because ISOBox is not just iterating over top boxes, it iterates over all boxes and it is also parsing payloads. However, for what we need here, it is enough with ensuring moov/mdat boxes are received so we don't put anything that breaks browser decoding process. 

I am going to do further testing using this new approach.

Reply:
@epiclabsDASH  - if we mainly use ISOBox for retrieving EMSG messages, then would your new parser be a replacement for that, assuming we added a component to it to parse only the EMSG payload?

Reply:
No, with the addition of the new parser I was not thinking on replacing ISOBox completely, at least for this PR. ISOBox is also used by MSS module, and not just as a box parser, it is also used to create boxes to ensure fragments have the structure required by the browser. So, replacing it completely will take time.

What I think makes sense is start replacing it in the cases in which we can get better performance doing something focused just on our requirements. ISOBox is a great library correctly designed for a generic purpose but, for example, for retrieving EMSG payload I think using it is a sledgehammer to crack a nut. We really don't need parsing all the boxes of a chunk to get payload of one (or more) of them. We can do much, much faster if we implement a parser that does just what we need. 

So, as a summary, in the short term we won't replace ISOBox completely. In the short term we will move parsing stuff that affects performance (EMSG boxes) to this new parser.